src/tmpfiles/tmpfiles.c

```diff
@@ -1024,19 +1041,15 @@ static int write_one_file(Item *i, const char *path) {
         }
 
         if (i->argument) {
-                _cleanup_free_ char *unescaped = NULL, *replaced = NULL;
+                _cleanup_free_ char *unescaped = NULL;
 
                 log_debug("%s to \"%s\".", i->type == CREATE_FILE ? "Appending" : "Writing", path);
 
                 r = cunescape(i->argument, 0, &unescaped);
                 if (r < 0)
                         return log_error_errno(r, "Failed to unescape parameter to write: %s", i->argument);
 
-                r = specifier_printf(unescaped, specifier_table, NULL, &replaced);
-                if (r < 0)
-                        return log_error_errno(r, "Failed to replace specifiers in parameter to write '%s': %m", unescaped);
-
-                r = loop_write(fd, replaced, strlen(replaced), false);
```

Comment:
so this change reverses the order in which we C-unescape and resolve specifiers now, and I think that's not OK. We should continue to resolve C-escapes first, and specifiers second I think. And that means you need to move the cunescape logic down, too into specifier_expansion_from_arg()

Reply:
Ok